django/lego/views.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Views."""
+
+from typing import Optional
+
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import PermissionDenied
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+
+from .dns import remove_dns_record, write_dns_record
+from .forms import CleanupForm, PresentForm
+from .models import Domain, DomainUserPermission
+
+
+@login_required
+def handle_present(request: HttpRequest) -> Optional[HttpResponse]:
+    """Handle the submissing of the present form.
+
+    Args:
+        request: the HTTP request.
+
+    Returns:
+        an HTTP response.
+
+    Raises:
+        PermissionDenied: if the user is not allowed to perform the operation.
+    """
+    if request.method == "POST":
+        form = PresentForm(request.POST)
+        if not form.is_valid():
+            return HttpResponse(form.errors.as_json())
+        user = request.user
+        try:
+            domain = Domain.objects.get(fqdn=form.cleaned_data["fqdn"])
+        except Domain.DoesNotExist:
+            return HttpResponse(status=404)
```

Comment:
This may be debatable, but shouldn't it rather be a PermissionDenied if the domain is not found in the database, as my understanding is that the database objects only handle the permission model.

Reply:
It is certainly an option. What do you think @mthaddon ?

Reply:
We should confirm what [the httprequest-lego spec](https://go-acme.github.io/lego/dns/httpreq/) expects here, and could look also look at what [the client charm](https://github.com/canonical/httprequest-lego-k8s-operator) does/expects if the spec isn't clear.

Reply:
I agree that returning a 403 is better than returning a 404 here. It means we always return a 200 or a 403 for any domain based on permissions and we're not doing something different if that domain does or doesn't exist.

Reply:
Changed, then. Thanks for the input!